components/server/src/workspace/workspace-starter.ts

```diff
@@ -522,8 +522,12 @@ export class WorkspaceStarter {
                         );
                     }
                 } catch (err) {
+                    let reason: FailedInstanceStartReason = "startOnClusterFailed";
+                    if ("code" in err && err.code === grpc.status.RESOURCE_EXHAUSTED) {
```

Comment:
Helper for checking the error is ResourceExhausted? Would avoid the "magic" expression in 2 places